cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java

```diff
@@ -80,6 +80,8 @@
   // 2 minutes (5-3) after last job completion
   static final String CLUSTER_REUSE_THRESHOLD_MINUTES = "clusterReuseThresholdMinutes";
   static final int CLUSTER_REUSE_THRESHOLD_MINUTES_DEFAULT = 3;
+  static final String CLUSTER_IDLE_TTL_MINUTES = "idleTTL";
+  static final int CLUSTER_IDLE_TTL_MINUTES_DEFAULT = 10;
```

Comment:
Major: 10 minutes is too short. Make it at least an hour to be safe. Under heavy concurrent pipeline launch, it is possible there is a large gap between dataproc cluster being created and the job being launched like tens of minutes. A short TTL would incorrect delete dataproc cluster. 

Recommend starting with a high TTL like 6 hours or even 12 hours. Can tighten it in feature when needed.

Reply:
For larger TTLs we would need product approval as it would be customer-billed time. Is it expected to have this large delay between cluster provisioning and job starting? @masoud-io could you comment? I mean it's a major problem by itself as this would be billed wasted time

Reply:
Per offline discussion, agreed on defaulting to 30mins in this PR.

Reply:
Created a JIRA to reduce it in future as discussed, https://cdap.atlassian.net/browse/CDAP-18707